lib/Serialization/ModuleFileSharedCore.cpp

```diff
@@ -370,12 +370,20 @@ static ValidationInfo validateControlBlock(
       if (forcedDebugRevision ||
           (requiresRevisionMatch && version::isCurrentCompilerTagged())) {
         StringRef compilerRevision = forcedDebugRevision ?
-          forcedDebugRevision : version::getCurrentCompilerTag();
+          forcedDebugRevision : version::getCurrentCompilerSerializationTag();
         if (moduleRevision != compilerRevision) {
-          result.status = Status::RevisionIncompatible;
+          // The module versions are mismatching, record it and diagnose later.
+          result.problematicRevision = moduleRevision;
```

Comment:
Could go with `mismatchedRevision`, but `problematic` is okay too

Reply:
I was working in diagnostics at the same time where we do use the generic `problematic` but here `mismatched` would be more precise. I'll try to update it next time I work in this code.